src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -292,6 +292,20 @@ public static void check(
     }
   }
 
+  private static int findMinArgsCounts(List<HaxeFunctionArgument> argumentList) {
+    int count = 0;
+    for (HaxeFunctionArgument argument : argumentList) {
+      if (argument.getOptionalMark() == null) {
+        if (!isVoidArgument(argument)) count++;
+      }
+    }
+    return count;
+  }
+
+  private static boolean isVoidArgument(HaxeFunctionArgument argument) {
+    return !(argument.getTypeOrAnonymous() == null  ||argument.getTypeOrAnonymous().getType() == null  || !argument.getTypeOrAnonymous().getType().getText().equals("Void"));
```

Comment:
NPE: getText() can return null, if the text element is not found.

This line of code is very obtuse, as well -- I had to seriously _think_ about it to understand it.  Maybe:
```Java
  HaxeTypeOrAnonymous toa = argument.getTypeOrAnonymous();
  HaxeType t = null != toa ? toa.getType() : null;
  String name = t != null ? t.getText() : null;
  return SpecificHaxeClassReference.VOID.equals(name);
```
I'd also rather not scatter references to type names all over the code base.
(Which also makes me think that if somebody actually uses the name "Void" for their own type, the name check would not be enough.)

Reply:
Is there a reason why line 2 of the requested change is `null != toa` and not `toa != null`?
Is the order something special?

Reply:
@CanIGetaPR It's an old habit that comes from coding in C.  You put the constant on the left of any comparison so that it's harder to make an unintended assignment.

For example, `if (myVar = null)` would be an assignment.  Now, in java, that is not a boolean expression, so the compiler would complain.  However `if (myBool = true)` is also an assignment and the Java compiler _doesn't_ complain about that.  So, it's just a good habit to train your mind to write them "backwards," particularly if you write in languages that allow an assignment in such places without issuing a warning.

Reply:
And, also, while I try to be consistent, I'm not always, as you can tell, because line 3 of the requested change isn't "backward."  
According to my rule, that line should have been `String name = null != t ? t.getText() : null;`.  Mea culpa.  😄

Reply:
Haha yes! I was confused why they were different, that caught my eye.